vaadin-date-picker-flow/src/main/java/com/vaadin/flow/component/datepicker/DatePicker.java

```diff
@@ -474,6 +498,13 @@ public LocalDate getInitialPosition() {
     @Override
     public void setRequired(boolean required) {
         super.setRequired(required);
+        this.required = required;
+    }
+
+    @Override
+    public void setRequiredIndicatorVisible(boolean requiredIndicatorVisible) {
+        super.setRequiredIndicatorVisible(required);
+        this.required = requiredIndicatorVisible;
```

Comment:
This could a breaking change as we are changing the behaviour of public API.
If somebody was using this indicator without setting required to the field (if there is such a case at all), it will be broken now.

Reply:
Seems to be doing the same thing.